lib/src/utils.dart

```diff
@@ -43,3 +46,122 @@ class MultipleTapGestureRecognizer extends TapGestureRecognizer {
     }
   }
 }
+
+/// This class is a placeholder class so that the renderer can distinguish
+/// between WidgetSpans and TextSpans
+class CustomTextSpan extends WidgetSpan {
```

Comment:
But if this is always a WidgetSpan, does this not go straight into the problem that inline spans are not - well - inline any more?

Reply:
I'm not super familiar with the technicalities of the `InlineSpan` in Flutter, but at least visually, there was no difference with this implementation. I included a parameter on this class for the original `InlineSpan` that would have been rendered beforehand. This parameter is used especially when rendering the `InteractableElement`s, so the original rendering is preserved as much as possible.

The class was a workaround just so I could use a `Builder` widget when rendering each span, that way I could add the `buildContext` to the Map that handles the list of possible contexts when a link is pressed.